alerting/provider/opsgenie/opsgenie_test.go

```diff
@@ -0,0 +1,331 @@
+package opsgenie
+
+import (
+	"github.com/TwiN/gatus/v3/alerting/alert"
+	"github.com/TwiN/gatus/v3/client"
+	"github.com/TwiN/gatus/v3/core"
+	"github.com/TwiN/gatus/v3/test"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestAlertProvider_IsValid(t *testing.T) {
+	invalidProvider := AlertProvider{APIKey: ""}
+	if invalidProvider.IsValid() {
+		t.Error("provider shouldn't have been valid")
+	}
+	validProvider := AlertProvider{APIKey: "2f8fe136-27a6-4a79-ab5b-a13cb0e59699"}
```

Comment:
Just making sure, this isn't your production API key, right? 🤣

Reply:
No 🤣
It's a random uuid, I will put a placeholder to keep it clean 👍🏼